### PR TITLE
Add hnswlib

### PR DESCRIPTION
**Pull request by junjiejiangjjj**

### CODE REVIEW

File: gptcache/manager/vector_data/hnswlib_store.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        np_data = np.array(data).astype("float32")
+        self._index.add_items(np_data, np.asarray(keys))
+
+    def search(self, data: "ndarray"):
+        np_data = np.array(data).astype("float32").reshape(1, -1)
+        ids, dist = self._index.knn_query(data=np_data, k=self._top_k)
+        return list(zip(dist[0], ids[0]))
+
+    def clear_strategy(self):
+        return ClearStrategy.DELETE
```
Comment (SimFG) [inline gptcache/manager/vector_data/hnswlib_store.py:41]: it should be `REBUILD`

File: gptcache/manager/vector_data/manager.py [hunk 2]
```diff
@@ -86,6 +86,17 @@ def get(name, **kwargs):
                 collection_name=collection_name,
                 top_k=top_k,
             )
+        elif name == "hnswlib":
+            from gptcache.manager.vector_data.hnswlib_store import Hnswlib
+
+            dimension = kwargs.get("dimension", DIMENSION)
+            index_path = kwargs.pop("index_path", "./hnswlib_index.bin")
+            max_elements = kwargs.pop("max_elements")
```
Comment (SimFG) [inline gptcache/manager/vector_data/manager.py:94]: should associate with the current `max_size` param

File: gptcache/manager/vector_data/hnswlib_store.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        self._index_file_path = index_file_path
+        self._dimension = dimension
+        self._max_elements = max_elements
+        self._index =  hnswlib.Index(space="l2", dim=self._dimension)
+        self._top_k = top_k
+        if os.path.isfile(self._index_file_path):
+            self._index.load_index(self._index_file_path, max_elements=max_elements)
+        else:
+            self._index.init_index(max_elements=max_elements, ef_construction=100, M=16)
+        self._index.set_ef(self._top_k * 2)
```
Comment (SimFG) [inline gptcache/manager/vector_data/hnswlib_store.py:25]: when `load_index`, it's not required `set_ef`.

Comment (junjiejiangjjj) [reply gptcache/manager/vector_data/hnswlib_store.py]: fixed

File: gptcache/manager/vector_data/hnswlib_store.py (hunk 1 above)
Comment (junjiejiangjjj) [reply gptcache/manager/vector_data/hnswlib_store.py]: fixed

File: gptcache/manager/vector_data/manager.py (hunk 2 above)
Comment (junjiejiangjjj) [reply gptcache/manager/vector_data/manager.py:94]: According to the current interface design, users can directly create a vector store, so it is not possible to associate max_size with it.

**Pull request merged**